source/containers.tex

```diff
@@ -15032,3 +15034,2422 @@
+    template<class OtherSizeType>
+      constexpr mapping(const extents_type&, const array<OtherSizeType, rank_>&) noexcept;
+
+    template<class StridedLayoutMapping>
+      constexpr explicit(@\seebelow@) mapping(const StridedLayoutMapping&) noexcept;
+
+    constexpr mapping& operator=(const mapping&) noexcept = default;
+
+    // \ref{mdspan.layout.stride.obs}, observers
+    constexpr const extents_type& extents() const noexcept { return @\exposid{extents_}@; }
+    constexpr array<size_type, rank_> strides() const noexcept
+    { return @\exposid{strides_}@; }
+
+    constexpr size_type required_span_size() const noexcept;
+
+    template<class... Indices>
+      constexpr size_type operator()(Indices...) const noexcept;
+
+    static constexpr bool is_always_unique() noexcept { return true; }
+    static constexpr bool is_always_contiguous() noexcept { return false; }
+    static constexpr bool is_always_strided() noexcept { return true; }
+
+    static constexpr bool is_unique() noexcept { return true; }
+    constexpr bool is_contiguous() const noexcept;
+    static constexpr bool is_strided() noexcept { return true; }
+
+    constexpr size_type stride(rank_type i) const noexcept { return @\exposid{strides_}@[i]; }
+
+    template<class OtherMapping>
+      friend constexpr bool operator==(const mapping&, const OtherMapping&) noexcept;
+
+  private:
+    extents_type @\exposid{extents_}@{};                    // \expos
+    array<size_type, rank_> @\exposid{strides_}@{};         // \expos
+  };
+}
+\end{codeblock}
+
+\pnum
+If \tcode{Extents} is not a specialization of \tcode{extents},
+then the program is ill-formed.
+
+\pnum
+\tcode{layout_stride::mapping<E>} is a trivially copyable type
+that models \tcode{regular} for each \tcode{E}.
+
+\rSec4[mdspan.layout.stride.expo]{Exposition-only helpers}
+
+\pnum
+Let \tcode{\exposid{REQUIRED-SPAN-SIZE}(e, strides)} be:
+\begin{itemize}
+\item
+1, if \tcode{e.rank() == 0} is \tcode{true}, otherwise
+\item
+0, if the size of the multidimensional index space \tcode{e} is 0, otherwise
+\item
+1 plus the sum of products of \tcode{(e.extent($r$) - 1)} and (strides[$r$])
+for all $r$ in the range $[0, \tcode{e.rank()})$.
+\end{itemize}
+
+\pnum
+Let \tcode{\exposid{OFFSET}(m)} be:
+\begin{itemize}
+\item
+\tcode{m()}, if \tcode{e.rank() == 0} is \tcode{true}, otherwise
+\item
+0, if the size of the multidimensional index space \tcode{e} is 0, otherwise
+\item
+\tcode{m(z...)} for a pack of integers \tcode{z}
+that is a multidimensional index into \tcode{m.extents()} and
+each element of \tcode{z} equals 0.
+\end{itemize}
+
+\pnum
+Let \exposid{is-extents} be the exposition-only variable template
+defined as follows:
+\begin{codeblock}
+template<class T>
+  constexpr bool @\exposid{is-extents}@ = false;                              // \expos
+template<class SizeType, size_t ... args>
+  constexpr bool @\exposid{is-extents}@<extents<SizeType, args...>> = true;   // \expos
```

Comment:
```suggestion
template<class SizeType, size_t... Args>
  constexpr bool @\exposid{is-extents}@<extents<SizeType, Args...>> = true;   // \expos
```

Reply:
No, these are values, not types, thus should be lowercase.

Reply:
`index_sequence` uses `I` rather than `i`. It's my understanding that template parameter names are camel case.

Reply:
Both `index_sequence` and `array` use capital letters for the non-type template parameters, so let's stick with that. Please open an issue if you want to reconsider this.